src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +104,64 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
         commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def runs = project.nunit.getTestInputAsList(getWhereFilters())
+
+        GParsPool.withPool {
+            runs.eachParallel {
+                def fileName = getRandomFileName()
+                logger.info("Filename generated for the \'$it\' input was \'$fileName\'")
+
+                def execArgs = project.nunit.buildCommandArgs(it, new File(intermediateReportsPath, fileName + ".xml"))
+                commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${execArgs.collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
+                commandLineArgs += "-output:${new File(intermediateReportsPath, fileName + ".xml")}"
+
+                execute(commandLineArgs)
+            }
+        }
+
+        execute(getMergeCommand(intermediateReportsPath))
```

Comment:
will this try to merge in case of failure of one of the tasks?

Reply:
As far as I've tested, yes. As we have for Nunit, for OpenCover we'll have ignoreFailures = true so the process will be closed successfully even if we have failing tests so it will get to the point of merging existing files